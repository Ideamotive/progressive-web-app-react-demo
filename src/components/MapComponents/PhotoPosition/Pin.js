import React from 'react';
import styled from 'styled-components';

export default styled.div`
    width: 60px;
    height: 60px;
    border-radius: 30px;
    border: 2px ${props => props.theme.colors.success} solid;
    background-size: cover;
    background-image: ${props => props.backgroundImage};
    transform: ${props => props.transform};
  `;