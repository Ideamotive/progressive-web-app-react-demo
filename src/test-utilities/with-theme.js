import React from 'react';
import { ThemeProvider, } from 'styled-components';

import theme from '../components/theme';

export const withTheme = (Component) => {
  return (props) => (
    <ThemeProvider theme={theme}>
      <Component {...props}/>
    </ThemeProvider>
  );
};
