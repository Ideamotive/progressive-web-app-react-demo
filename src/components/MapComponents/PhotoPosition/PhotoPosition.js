import React from 'react';
import PropTypes from 'prop-types';

// Styled Components
import Pin from './Pin';

const PhotoPosition = (props) => {
  const { onPhotoClick, imgUrl, zoom, id } = props;
  const scale = ((zoom - 4) / 15);
  return (
    <Pin
      onClick = {() => {onPhotoClick(id)}}
      backgroundImage={ `url(${imgUrl})`}
      transform={ `translate(-30px, -30px) scale(${scale})`}
    />
  );
};

PhotoPosition.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  zoom: PropTypes.number.isRequired,
};

PhotoPosition.defaultProps = {
  onClick: () => null,
  zoom: 10,
};

export default PhotoPosition;