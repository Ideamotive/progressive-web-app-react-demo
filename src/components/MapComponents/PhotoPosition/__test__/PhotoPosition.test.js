import React from 'react';
import * as renderer from 'react-test-renderer';
import { withTheme } from '../../../../test-utilities/with-theme';
import PhotoPosition from '../PhotoPosition';

describe('<PhotoPosition/> component', () => {
  test('should match to snapshot', () => {
    const PhotoPositionWithTheme = withTheme(PhotoPosition);
    const tree = renderer
      .create(<PhotoPositionWithTheme
        imgUrl={'http://test.test'}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

