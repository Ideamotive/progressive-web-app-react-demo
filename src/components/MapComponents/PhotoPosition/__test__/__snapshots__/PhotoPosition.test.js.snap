// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoPosition/> component should match to snapshot 1`] = `
<div
  className="sc-bdVaJa jysxDn"
  onClick={[Function]}
  transform="translate(-30px, -30px) scale(0.4)"
/>
`;
